src/qamqpclient.cpp

```diff
@@ -660,14 +660,20 @@ QAmqpExchange *QAmqpClient::createExchange(int channelNumber)
 QAmqpExchange *QAmqpClient::createExchange(const QString &name, int channelNumber)
 {
     Q_D(QAmqpClient);
+    if (!name.isEmpty() && d->exchanges.contains(name)
+                    && (!d->exchanges[name].isNull()))
```

Comment:
you might find some inspiration from the [QObjectCleanupHandler](http://code.woboq.org/qt5/qtbase/src/corelib/kernel/qobjectcleanuphandler.cpp.html), as an easy way to refactor this functionality into a small utility class for tracking QAmapChannels - just a thought